node/network/bridge/src/action.rs

```diff
@@ -182,3 +198,27 @@ impl From<Option<NetworkEvent>> for Action {
 		}
 	}
 }
+
+impl From<Option<Result<AllMessages, RequestMultiplexError>>> for Action {
+	fn from(event: Option<Result<AllMessages, RequestMultiplexError>>) -> Self {
+		match event {
+			None => {
+				tracing::info!(
+					target: LOG_TARGET,
+					"Shutting down Network Bridge: underlying request stream concluded"
+				);
+				Action::Abort
+			}
+			Some(Err(err)) => {
+				tracing::debug!(
+					target: LOG_TARGET,
+					"Peer '{}' sent us an invalid request: {}",
+					err.peer,
+					err.error
+				);
+				Self::ReportPeer(err.peer, MALFORMED_MESSAGE_COST)
+			}
+			Some(Ok(msg)) => Self::SendMessage(msg),
+		}
```

Comment:
A `From` trait implementation having side-effects is really a high level of hacky-ness.

Reply:
Agreed! Thanks! *runningandfixingthisinshame*

Reply:
Wait! What side effects are you talking about - the tracing?

Reply:
Ok less severe, than I initially thought, but still kinda ugly. Easily fixable though, will do.